src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressParsingTest.cs

```diff
@@ -163,7 +165,13 @@ public void TestLargeListOfValidEmails_WithInvalidEmails_ShouldThrow()
         [MemberData(nameof(GetInvalidEmailTestData))]
         public void TestInvalidEmailAddresses(string address)
         {
-            Assert.Throws<FormatException>(() => { new MailAddress(address); });
+            Action act = () => new MailAddress(address);
+            if (address is null)
+                Assert.Throws<ArgumentNullException>(act);
+            else if (address == string.Empty)
+                Assert.Throws<ArgumentException>(act);
+            else
+                Assert.Throws<FormatException>(act);
```

Comment:
Just a personal taste, definitely won't hold the PR on this:
```suggestion
            Action act = () => new MailAddress(address);
            if (address is null)
            {
                Assert.Throws<ArgumentNullException>(act);
            }
            else if (address == string.Empty)
            {
                Assert.Throws<ArgumentException>(act);
            }
            else
            {
                Assert.Throws<FormatException>(act);
            }
```